New cloak added to the tg database

Comxy opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
I now know what github is all about, and want to improve the quality of our game as such. I have added it before to the game, but spawning it in is really easy and not rewarding. This is why, I added a cloak that rarely spawns into the maints and is craftable.
## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
It benefits this game a whole lot. A lot of people have been playing for some time now. This drip cloak gives people something to search for, or a new objective to build a drip cloak. As you can see, making a drip cloak isn't the easiest thing to do. It will also be decently difficult to find it in the maints. That is why this would be a good new end-game like additions to our station.
## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. -->

:cl:
add: Added a new obtainable drip cloak found in maintenance or craftable.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Comxy committed:
I added my own cloak to cloaks.dm

This is more like a test. I will also try adding my cloak to the icon files, and then hope it works and goes through. I wanted to add my own clothing through the game with icon files, but it seems to be hardwired in the code.

Comxy committed:
Replaces the cloaks with a new version

This also adds my cloak

Comxy committed:
Adding my cloak to the neck.dmi file

It adds my cloak to the other location where the cloak is hard wired coded to reference to.

Comxy committed:
Merge pull request #1 from Comxy/comxy-cloak-path

I added my own cloak to cloaks.dm #add

JohnFulpWillard commented:
I fear this PR's raw energy

TheBonded commented:
If you wanted this for your own personal server, why not just merge this to your own split branch and run that code instead? It would just be a copy of tg's code but with the cloak added

Nobelium-XVIII commented:
The longer he's on GitHub the more powerful he becomes...

Comxy commented:
> 
> 
> If you wanted this for your own personal server, why not just merge this to your own split branch and run that code instead? It would just be a copy of tg's code but with the cloak added

There are three reasons I am trying to get this on the main branch. This is because:

1:
I don't play on my own server; I have never started up my own server, nor ever tried. To me seeing other people play or playing myself is to me way more fun. Also, I don't have much to test if I do start my own local server.

2:
This is a test thus maybe an excuse to add my own drip to the game. Everyone has to start somewhere with adding changes and this is a small thing, perfect for someone new to github branches. 

3:
I tried to add this with changing item icons, but the sad truth is that the icons are hardwired in the code using parent types.

TheBonded commented:
> > 
> > 
> > If you wanted this for your own personal server, why not just merge this to your own split branch and run that code instead? It would just be a copy of tg's code but with the cloak added
> 
> There are three reasons I am trying to get this on the main branch. This is because:
> 
> 1:
> I don't play on my own server; I have never started up my own server, nor ever tried. To me seeing other people play or playing myself is to me way more fun. Also, I don't have much to test if I do start my own local server.
> 
> 2:
> This is a test thus maybe an excuse to add my own drip to the game. Everyone has to start somewhere with adding changes and this is a small thing, perfect for someone new to github branches. 
> 
> 3:
> I tried to add this with changing item icons, but the sad truth is that the icons are hardwired in the code using parent types for probably most clothing, including cloaks.

In that case, you might wish to change the changelog to administrative or such to not suffer the gbp loss of adding something unobtainable

wesoda25 commented:
Can you add a cloak for me while you're at it?

Comxy commented:
> 
> 
> Can you add a cloak for me while you're at it?

That is honestly a good question too hard for me to answer. But I would like to say that it would be too hard for me to sprite and poll request a cape for every person as selfish as that may sound.

MrDoomBringer commented:
walks in adds his own drip 
what a fucking based PR
merge it

TheBonded commented:
make a check that gives you this cloak if you have your (@Comxy's) ckey so you always get it whenever you join a round

Mothblocks commented:
this is honestly so powerful that i don't know if it's within my jurisdiction to close this

D4C-420 commented:
a 21 tc trade of a pr.

Krysonism reviewed (approved):
Sprite looks good!

optimumtact commented:
An unobtainable cloak is not something we need, try to add it to a ruin, or make it in a vendor or anything.

Mothblocks commented:
and then call it something other than ur ckey

MrDoomBringer commented:
> and then call it something other than ur ckey

Some random cloak with worse stats then shift-start-available gear is not what this game needs
A commentary on the futility of player references, however? We could use that

Timberpoes commented:
Also there is a way to var edit this to work in game. Someone will beat me to it but clothing has override vars for just this occasion. But I have definitely created custom var edited cloaks in the past.

Comxy committed:
Merge branch 'tgstation:master' into master

KittyNoodle commented:
Speedmerge this asap, the longer it takes the stronger he becomes

Comxy committed:
A name patch and new drip cloak

It is actually a crime for people to not know about item lore while wearing something, so I took the lore out of the name

Comxy committed:
Merge pull request #2 from Comxy/drip-cloak-path

A name patch and new drip cloak

LemonInTheDark commented:
I vote we toss him in a pit of lava

Comxy committed:
Added the drip cloak to the maint loot spawner, as a rare clothing item

The cloak drip is now obtainable

Comxy committed:
Merge pull request #3 from Comxy/drip_cloak_path_maintloot

Added the drip cloak to the maint loot spawner, as a rare clothing item

Comxy committed:
New drip cloak recipe

For if it sadly doesn't spawn in maintenance areas, you will have to do the craftsmanship yourself. You will need 8 different items as listed in the code, and will be listed in the crafting menu automatically hopefully. The cloak is complex thus it will take a while.

Comxy committed:
Merge pull request #4 from Comxy/drip-cloak-crafting-recipe

New drip cloak recipe

Watermelon914 commented on code/modules/clothing/suits/cloaks.dm:
```diff
@@ -48,6 +48,13 @@
 	desc = "Worn by the Head of Personnel. It smells faintly of bureaucracy."
 	icon_state = "hopcloak"
+	
+/obj/item/clothing/neck/cloak/comxy
```
rename type to something other than comxy

Watermelon914 commented on code/modules/clothing/suits/cloaks.dm:
```diff
@@ -48,6 +48,13 @@
+/obj/item/clothing/neck/cloak/comxy
+	name = "drip cloak"
+	desc = "A cloak that makes you cooler than other people and it has a label called comxia on the inside of the cloak."
+	icon_state = "comxycloak"
```
remove the player reference from the icon state and the description

Comxy committed:
Name change

Yeet

Comxy committed:
drip

yeet

Comxy committed:
Drip

Yeet

Comxy committed:
cloakname change

yeet

Comxy committed:
another DMI name change

yeet

Comxy committed:
Merge pull request #5 from Comxy/drip-cloak-name-path

Name change

Comxy committed:
Merge pull request #6 from Comxy/comxy-drip-maint-change

drip

Comxy committed:
Merge pull request #7 from Comxy/drip-cloak-name-change

Drip

Comxy committed:
Merge pull request #8 from Comxy/another-name-change

cloakname change

Comxy committed:
Merge pull request #9 from Comxy/path-of-drip

another DMI name change

Comxy committed:
Revert "drip"

Comxy committed:
Revert "Name change"

Comxy committed:
Merge pull request #11 from Comxy/revert-6-comxy-drip-maint-change

Revert "drip" for a reload, I published the wrong ones first

Comxy committed:
Merge pull request #12 from Comxy/revert-5-drip-cloak-name-path

Revert "Name change" for a reload, I published the wrong ones first

Comxy committed:
comxy to drip

yeet

Comxy committed:
pull revert

Comxy committed:
comxy to drip unrevert

Comxy committed:
Revert "Revert "drip" for a reload, I published the wrong ones first"

Comxy committed:
Merge pull request #13 from Comxy/comxy-to-drip

comxy to drip unrevert

Comxy committed:
Merge pull request #14 from Comxy/revert-11-revert-6-comxy-drip-maint-change

weird github stuff

Comxy commented:
Alright, I tried reverting the initial name changes, since it checked the preupdated ones the drip objects was still called the comxy object, everything should be fine, I tried rereverting but I am not sure what is going on.

Comxy committed:
another drip update

This fixed the last issues I think and hope

Comxy committed:
Merge branch 'tgstation:master' into master

Comxy committed:
Merge pull request #15 from Comxy/comxy-drip-path

another drip update

Watermelon914 commented:
please install visual studio code and use that to code instead of webedits. You need to properly test PRs to make sure they work.

LemonInTheDark commented:
I swear to god if we add a drip cloak to match the drip jacket I'm going to break.

Tennessee116 commented:
This is the moment comxy became the master of the drip.

Geevies commented:
oranges merge this

Comxy committed:
vsc test

Comxy committed:
yeet

maxymax13 commented:
>adds his own clothing item
>does not elaborate
chad webeditor

Comxy committed:
Merge branch 'tgstation:master' into master

Comxy committed:
Merge branch 'tgstation:master' into master

Comxy commented:
I will just close this before it auto closes. Personally I thought the sprite wasn't that good, so I see why it wasn't merged. The animation when the light disappeared didn't have a good transition. Also the recipe was probably too big, but I have learned from my mistakes, soon, a new one will open up one day sooner or later.

LemonInTheDark commented:
It's like, we don't really like self references that are as clear as you're doing. Speaking personally I don't really like stuff like the drip outfit, think the aesthetic is a bit off.

MrDoomBringer commented:
we were so close to greatness
